MachObfuscator/Commons/Paths.swift

```diff
@@ -2,8 +2,50 @@ import Foundation
 
 enum Paths {
     static let separator: Character = "/"
-    static let macosFrameworksRoot = "/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk"
-    static let iosFrameworksRoot = "/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS.sdk"
+
+    static func xcode_select() -> String? {
+        let task = Process()
+
+        task.executableURL = URL(fileURLWithPath: "/usr/bin/xcode-select")
+        task.arguments = ["--print-path"]
+        let outputPipe = Pipe()
+        task.standardOutput = outputPipe
+        try! task.run()
+        task.waitUntilExit()
+        guard task.terminationStatus == 0 else {
+            LOGGER.warn("Unable to invoke xcode-select")
+            return nil
+        }
+
+        let outputData = outputPipe.fileHandleForReading.readDataToEndOfFile()
+        let output = String(decoding: outputData, as: UTF8.self)
+        let path = output.trimmingCharacters(in: .newlines)
+
+        // Test just in case, protects also against unexpected output
+        guard FileManager.default.fileExists(atPath: path) else {
+            LOGGER.warn("xcode-select returned non-existing path: \(path)")
+            return nil
+        }
+
+        return path
+    }
+
+    static let xcodeRoot: String = {
+        let root =
+            // If running unit tests DEVELOPER_DIR variable should be set.
+            // DEVELOPER_DIR environment variable may also be used to set specific directory.
+            ProcessInfo.processInfo.environment["DEVELOPER_DIR"] ??
```

Comment:
Do you think we need this `DEVELOPER_DIR` explicitly? `xcode-select` recognises this environment variable.

Reply:
Changed. I was not aware that `xcode-select` takes `DEVELOPER_DIR` into account.